src/Configuration/src/Kubernetes.ServiceBinding/ServiceBindingConfigurationProvider.cs

```diff
@@ -0,0 +1,282 @@
+
+        if (!_source.IgnoreKeyPredicate(secretkey))
+        {
+            data[secretkey] = secretEntry.Value;
+        }
+    }
+
+    internal class ServiceBinding
+    {
+        private readonly Dictionary<string, string> _secrets;
+        public string Name { get; }
+
+        public string Path { get; }
+
+        public string Provider { get; }
+
+        public IDictionary<string, string> Secrets => new ReadOnlyDictionary<string, string>(_secrets);
+
+        public string Type { get; }
+
+        // Creates a new Binding instance using the specified file system directory
+        public ServiceBinding(string path)
+            : this(System.IO.Path.GetFileName(path), path, CreateSecrets(path))
+        {
+        }
+
+        // Creates a new Binding instance using the specified content
+        public ServiceBinding(string name, string path, IDictionary<string, string> secret)
+        {
+            Name = name ?? throw new ArgumentException("Binding has no name and is not a valid binding");
+            Path = path ?? throw new ArgumentException("Binding has no path and is not a valid binding");
+
+            _secrets = new Dictionary<string, string>();
+
+            string type = null;
+            string provider = null;
+
+            foreach (KeyValuePair<string, string> entry in secret)
+            {
+                switch (entry.Key)
+                {
+                    case TypeKey:
+                        type = entry.Value;
+                        break;
+                    case ProviderKey:
+                        provider = entry.Value;
+                        break;
+                    default:
+                        _secrets.Add(entry.Key, entry.Value);
+                        break;
+                }
+            }
+
+            Type = type ?? throw new ArgumentException($"{path} has no type and is not a valid binding");
+            Provider = provider;
+        }
+
+        private static IDictionary<string, string> CreateSecrets(string path)
+        {
+            return CreateFilePerEntry(path);
+        }
+
+        private static IDictionary<string, string> CreateFilePerEntry(string path)
+        {
+            try
+            {
+                IEnumerable<string> files = Directory.EnumerateFiles(path);
+                var result = new Dictionary<string, string>();
+
+                if (!files.Any())
+                {
+                    return result;
+                }
+
+                foreach (string file in files)
+                {
+                    FileInfo fileInfo = new(file);
+
+                    if (fileInfo.Exists && !IsHidden(file))
+                    {
+                        result.Add(fileInfo.Name, ReadContentsAsString(file));
+                    }
+                }
+
+                return result;
+            }
+            catch (Exception)
+            {
+                // Log
+                throw new ArgumentException("${path} is invalid");
```

Comment:
```suggestion
                throw new ArgumentException($"Path '{path}' is invalid");
```
Looks like broken string interpolation, does this look like an improvement?